ec/resources/dnsroute/secondary_zone.go

```diff
@@ -155,117 +187,150 @@ func ResourceSecondaryZoneGroupCreate(ctx context.Context, d *schema.ResourceDat
-		ID:              -1,
+	secondaryZoneGroup := routedns.SecondaryZoneGroup{
 		Name:            name,
 		ZoneComposition: zcr,
-		CustomerID:      -1,
 	}
 
-	log.Printf("[INFO] Creating a new Secondary Zone Group for Account '%s': %+v", accountNumber, secondaryZoneGroupRequest)
-
-	dnsrouteClient := api.NewDNSRouteAPIClient(*config)
+	// Call Create Secondary Zone Group API
+	log.Printf(
+		"[INFO] Creating a new Secondary Zone Group for Account '%s': %+v",
+		accountNumber,
+		secondaryZoneGroup,
+	)
+	params := routedns.NewAddSecondaryZoneGroupParams()
+	params.AccountNumber = accountNumber
+	params.SecondaryZoneGroup = secondaryZoneGroup
 
-	resp, err := dnsrouteClient.AddSecondaryZoneGroup(secondaryZoneGroupRequest)
+	resp, err := routeDNSService.AddSecondaryZoneGroup(*params)
 
 	if err != nil {
+		d.SetId("")
 		return diag.FromErr(err)
 	}
 
-	log.Printf("[INFO] Create successful - New Secondary Zone Group ID: %d", resp)
-	d.SetId(strconv.Itoa(resp))
+	log.Printf(
+		"[INFO] Create successful - New Secondary Zone Group ID: %d",
+		resp.ID,
+	)
 
-	ResourceSecondaryZoneGroupRead(ctx, d, m)
+	d.SetId(strconv.Itoa(resp.ID))
 
-	return diags
+	return ResourceSecondaryZoneGroupRead(ctx, d, m)
 }
 
-func ResourceSecondaryZoneGroupRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	var diags diag.Diagnostics
-	secondaryZoneGroupId, err := strconv.Atoi(d.Id())
-	accountNumber := d.Get("account_number").(string)
+func ResourceSecondaryZoneGroupRead(
+	ctx context.Context,
+	d *schema.ResourceData,
+	m interface{},
+) diag.Diagnostics {
+	// Initialize Route DNS Service
+	secondaryZoneGroupID, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
 
+	accountNumber := d.Get("account_number").(string)
 	config := m.(**api.ClientConfig)
-	(*config).AccountNumber = accountNumber
+	routeDNSService, err := buildRouteDNSService(**config)
+	if err != nil {
+		return diag.FromErr(err)
+	}
 
-	dnsRouteClient := api.NewDNSRouteAPIClient(*config)
+	// Call Get Secondary Zone Group API
+	params := routedns.NewGetSecondaryZoneGroupParams()
+	params.AccountNumber = accountNumber
+	params.ID = secondaryZoneGroupID
 
-	log.Printf("[INFO] Retrieving Secondary Zone Group by zoneID: %d", secondaryZoneGroupId)
+	log.Printf(
+		"[INFO] Retrieving Secondary Zone Group by zoneID: %d",
+		secondaryZoneGroupID,
+	)
 
-	resp, err := dnsRouteClient.GetSecondaryZoneGroup(secondaryZoneGroupId)
+	resp, err := routeDNSService.GetSecondaryZoneGroup(*params)
 
 	if err != nil {
-		d.SetId("")
 		return diag.FromErr(err)
 	}
 
 	log.Printf("[INFO] Retrieved Secondary Zone Group: %+v", resp)
+
+	// Update Terraform state with retrieved Secondary Zone Group data
 	newID := strconv.Itoa(resp.ID)
 	d.Set("name", resp.Name)
 	d.Set("account_number", accountNumber)
+	//TODO: Properly process zone composition into terraform state
 	//d.Set("ZoneComposition", resp.ZoneComposition)
```

Comment:
Do we have a ticket for this TODO?

Reply:
I am not sure why this isn't rendering in the UI diff but if you look at the commit for the PR merged into this branch you'll see the change.

https://github.com/EdgeCast/terraform-provider-edgecast/pull/77/commits/b2404951c10da06dfdcf76a96fca0d1d74b9238a#diff-78bae89ee6828ba869232f4c687f496aa7232775049e04877c318193e85877c4R239

Done! Thank you!